### PR TITLE
Fixes HMR for ES modules

### PR DESCRIPTION
**Pull request by nddery**
Pass in `this` as `ctx` to the API, which reference the updated native ES module. There seems to be a reference to the module in `module.__proto__.exports.a` but (1) it seems weird to use that, and (2) it doesn't work with it as it doesn't have the `beforeCreate` and al. hooks from `vue-hot-reload-api`. Not sure `ctx` is the best name for that variable but.. that's what I got.

ref: #7

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -47,7 +48,7 @@ exports.default = function (_ref) {
   if (!module.exports) {
     // babel did not transform modules
     // eslint-disable-next-line no-underscore-dangle
-    component = module.__esModule ? module.default : module;
+    component = ctx.__esModule ? ctx.default : ctx;
```
Comment (nddery) [inline lib/api.js]: From my tests there was always a `default` property on the object (even if that's all that's being exported), so maybe we could just drop this and test for `default` ? Not sure what to do when there are no default and only named exports ?

**Comment by skyrpex** [comment]
Perfect! Now it really works. Thank you so much.

File: lib/api.js (hunk 1 above)
Comment (skyrpex) [inline lib/api.js]: That's a good question. I guess we should just take care of the default export (I think that's how vue-loader works, too).

Comment (nddery) [inline lib/api.js]: Yeah, you're right, that's what they (vue-loader) seem to [do](https://github.com/vuejs/vue-loader/blob/master/lib/loader.js#L480). But is it `*.vue` files that don't support named export or `vue-hot-reload-api` that doesn't like it or something ?

Comment (skyrpex) [inline lib/api.js]: I guess it's not supported because there's no way to determine what's a Vue object or something completely unrelated.

**Pull request merged**